tasks/delete_owner.py

```diff
@@ -1,118 +1,21 @@
 import logging
 
-from celery.exceptions import SoftTimeLimitExceeded
 from shared.celery_config import delete_owner_task_name
 
 from app import celery_app
-from database.models import Branch, Commit, LoginSession, Owner, Pull, Repository
-from database.models.core import CompareCommit
-from services.archive import ArchiveService
+from services.cleanup.owner import cleanup_owner
+from services.cleanup.utils import CleanupSummary
 from tasks.base import BaseCodecovTask
 
 log = logging.getLogger(__name__)
 
 
 class DeleteOwnerTask(BaseCodecovTask, name=delete_owner_task_name):
-    """
-    Delete an owner and their data:
-    - Repo archive data for each of their owned repos
-    - Owner entry from db
-    - Cascading deletes of repos, pulls, and branches for the owner
-    """
+    acks_late = True  # retry the task when the worker dies for whatever reason
```

Comment:
It won't retry if the task ends with an Exception... just saying, if that is the expectation in "whatever"

Reply:
Do you have more details about the execution model of celery? I feel like after all this time I still don’t understand how it works and behaves under various circumstances.

For these tasks, I would want them to "resume" until they are completed.
This means "resuming" on timeouts (not sure if you can configure them to not have a timeout at all?).
It means resuming when the worker is killed for whatever reason. It also means resuming on any kind of random infrastructure exception.

Even for a fixable programmer error, I would like these tasks to always run to "successful" completion.

I’m not entirely sure how to achieve that with celery 🤷🏻‍♂️